### PR TITLE
Implements Issue #842 OverrideHasInfiniteAmmo

### PR DESCRIPTION
**Pull request by Musashi1584**

### CODE REVIEW

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Item.uc [hunk 1]
```diff
@@ -952,8 +952,41 @@ simulated function int GetClipSize()
 	return ClipSize;
 }
 
+/// HL-Docs: feature:OverrideHasInfiniteAmmo; issue:842; tags:tactical
+/// Allows listeners to override the result of HasInfiniteAmmo
+///
+/// ```unrealscript
+/// EventID: OverrideProjectileInstance
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Item.uc:959]:
```suggestion
/// EventID: OverrideHasInfiniteAmmo
```

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Item.uc [hunk 2]
```diff
@@ -952,8 +952,41 @@ simulated function int GetClipSize()
 	return ClipSize;
 }
 
+/// HL-Docs: feature:OverrideHasInfiniteAmmo; issue:842; tags:tactical
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Item.uc:955]:
```suggestion
/// HL-Docs: feature:OverrideHasInfiniteAmmo; issue:842; tags:tactical,events
```

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Item.uc [hunk 3]
```diff
@@ -952,8 +952,41 @@ simulated function int GetClipSize()
...
+/// Allows listeners to override the result of HasInfiniteAmmo
+///
+/// ```unrealscript
+/// EventID: OverrideProjectileInstance
+/// EventData: XComLWTuple {
+///     Data: [
+///       out bool bOverrideHasInfiniteAmmo
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Item.uc:962]:
```suggestion
///       out bool bOverrideHasInfiniteAmmo,
```

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Item.uc [hunk 4]
```diff
@@ -952,8 +952,41 @@ simulated function int GetClipSize()
...
+///       out bool bHasInfiniteAmmo
+///     ]
+/// }
+/// EventSource: XComGameState_Item self
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Item.uc:966]:
```suggestion
/// EventSource: XComGameState_Item (self)
```

**Comment by Musashi1584** [comment]
@Xymanek 
i refactored the code. can you please re-review it?

**Pull request merged**